src/editor.c

```diff
@@ -782,8 +767,117 @@ static gboolean autocomplete_scope(GeanyEditor *editor, const gchar *root, gsize
 		current_scope = "";
 	tags = tm_workspace_find_scope_members(editor->document->tm_file, name, function,
 				member, current_scope, scope_sep_typed);
+
+	g_free(name);
+	return tags;
+}
+
+
+static GPtrArray *get_tags_for_current_scope(GeanyEditor *editor, const gchar *name, gboolean prefix)
+{
+	GPtrArray *found_tags = NULL;
+	const gchar *current_scope;
+
+	if (symbols_get_current_scope(editor->document, &current_scope) != -1)
+	{
+		const GeanyFiletype *const ft = editor->document->file_type;
+		const gchar *const context_sep = tm_tag_context_separator(ft->lang);
+		gchar *const scope = g_strdup(current_scope);
+		GPtrArray *tags;
+		TMTag *tag;
+		guint i;
+
+		found_tags = g_ptr_array_new();
+		while (TRUE)
+		{
+			gchar *const sep_pos = g_strrstr(scope, context_sep);
+			const gchar *const pass_name = sep_pos ? (sep_pos + strlen(context_sep)) : scope;
+			const gchar *const pass_scope = sep_pos ? scope : "";
+
+			if (sep_pos)
+				*sep_pos = 0;
+
+			tags = tm_workspace_find_scope_members(editor->document->tm_file, pass_name,
+						FALSE, FALSE, pass_scope, TRUE);
+			if (tags)
+			{
+				foreach_ptr_array(tag, i, tags)
+				{
+					if ((prefix && g_str_has_prefix(tag->name, name)) ||
+						(! prefix && strcmp(tag->name, name) == 0))
+					{
+						g_ptr_array_add(found_tags, tag);
+					}
+				}
+
+				g_ptr_array_free(tags, TRUE);
+			}
+
+			if (! sep_pos)
+				break;
+		}
+
+		g_free(scope);
+
+		/* root tags */
+		if (prefix)
+			tags = tm_workspace_find_prefix(name, ft->lang, 0xffff);
+		else
+			tags = tm_workspace_find(name, NULL, tm_tag_max_t, NULL, ft->lang);
```

Comment:
I later altered everything to manually dedup (in the autoc side) so I can chose depending on calltip/autoc